api/app/GraphQL/Mutations/UpdateUserSkillRankings.php

```diff
@@ -0,0 +1,92 @@
+<?php
+
+namespace App\GraphQL\Mutations;
+
+use App\Models\User;
+
+final class UpdateUserSkillRankings
+{
+    /**
+     * Update a user's UserSkill rankings
+     *
+     * @param  null  $_
+     * @param  array{userId: UUID, userSkillRanking: UpdateUserSkillRankingsInput}  $args
+     * @return User
+     */
+    public function __invoke($_, array $args)
+    {
+        $user = User::find($args['userId'])->load(['userSkills', 'userSkills.skill']);
+        $userSkillsCollection = $user->userSkills;
+        $userSkillRankingInput = $args['userSkillRanking'];
+
+        // execute blocks depending on whether the value at args.userSkillRankingInput.X is non-null
+        if (isset($userSkillRankingInput['topTechnicalSkillsRanked'])) {
+
+            // wipe existing technical top skills ranking
+            $arrayUserSkillId = $userSkillRankingInput['topTechnicalSkillsRanked'];
+            $userSkillsCollection->where('skill.category', 'TECHNICAL')
+                ->each
+                ->update(['top_skills_rank' => null]);
```

Comment:
Oh, yeah I see why. It is because collection and not an eloquent relationship. I guess it is just another subject one 😅 

This does works as well but I have no idea if either is preferred. What you have is working so let's stick with it 👍 

```php
// wipe existing technical top skills ranking
$user->userSkills()->whereHas('skill', function ($query) {
  return $query->where('category', 'TECHNICAL');
})->update(['top_skills_rank' => null]);
```

Reply:
Sounds good to me!